qiskit/quantum_info/states/statevector.py

```diff
@@ -369,6 +394,13 @@ def expectation_value(self, oper, qargs=None):
         Returns:
             complex: the expectation value.
         """
+        if isinstance(oper, Pauli):
+            return self._expectation_value_pauli(oper)
+
+        if isinstance(oper, SparsePauliOp):
+            return sum([coeff * self._expectation_value_pauli(Pauli(p))
+                        for p, coeff in zip(oper.table.to_labels(), oper.coeffs)])
```

Comment:
This can be more efficient by avoiding the label conversion
```python
return sum([coeff * self._expectation_value_pauli(Pauli((z, x))
                        for z, x, coeff in zip(oper.table.Z, oper.table.X, oper.coeffs)])
```